### ISSUE TITLE
Develop test script

### ISSUE THREAD

**Issue by jgoldfar** [robypoteau/qlopt#3]
The makefile references a test script and corresponding test files. Were these ever completed, and/or should they be added?

**Comment by jgoldfar** [comment]
Follow up given the results of [the CI build](https://travis-ci.org/jgoldfar/qlopt/builds/481532370): the executables error out on Linux platforms, but not on OSX. This is independent of the compiler, though we should check the Makefile isn't overriding the compiler.

**Comment by jgoldfar** [comment]
So:
- The CC variable in the makefile was overriding the environment's compiler setting, but
- the segfault on ubuntu seems real; I can [reproduce it](https://gist.github.com/b4863945d02839f5276ea3e2feb96bac) on a local Ubuntu server, if any collaborators would like access, let me know) but it's not in the core computational code, so should be easy to fix.
- Seems unlikely, but possible that segfault is due to use of Eigen 3.3.7, which is not the version documented to work...

**Comment by robypoteau** [comment]
> The makefile references a test script and corresponding test files. Were these ever completed, and/or should they be added?

No, I never ended up creating any test. Also yes they should be added, that was on my todo list.

**Comment by robypoteau** [comment]
> Follow up given the results of [the CI build](https://travis-ci.org/jgoldfar/qlopt/builds/481532370): the executables error out on Linux platforms, but not on OSX. This is independent of the compiler, though we should check the Makefile isn't overriding the co

> Follow up given the results of [the CI build](https://travis-ci.org/jgoldfar/qlopt/builds/481532370): the executables error out on Linux platforms, but not on OSX. This is independent of the compiler, though we should check the Makefile isn't overriding the compiler.

I made the CXX fix. Also, I think some of the errors your seeing with the algorithm converging is because of some of the refactoring I did. I was focused on the ThreeStep... example and didn't make changes to the LotkaVolterra examples until later.

**Comment by jgoldfar** [comment]
Indeed, the code that was available was the problem. The convergence issues actually presented in all example files, but with your update, things now work on new-enough Debian-based linux distributions.
